connectivity/suite.go

```diff
@@ -195,9 +195,14 @@ var (
 	client2Label = map[string]string{"name": "client2"}
 )
 
-func Run(ctx context.Context, ct *check.ConnectivityTest, addExtraTests func(*check.ConnectivityTest) error,
-	addExtraSetup func(context.Context, *check.ConnectivityTest) error) error {
-	if err := ct.SetupAndValidate(ctx, addExtraSetup); err != nil {
+type hooks interface {
+	DetectFeatures(ctx context.Context, ct *check.ConnectivityTest) error
```

Comment:
Wondering if we could  compose off interface defined above?

nit: Docs on ifc methods